src/components/Api.js

```diff
@@ -0,0 +1,155 @@
+export default class Api {
+  constructor(options) {
+    this._url = options.baseUrl;
+    this._headers = options.headers;
+    // this._test =
+  }
+
+  getUserInfo() {
+    return fetch(`${this._url}/users/me`, {
+      headers: this._headers,
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+
+  setUserInfo(values, submitButton) {
+    submitButton.textContent = "Saving ...";
+    return fetch(`${this._url}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify(values),
+    })
+      .then((res) => {
+        if (res.ok) {
+          submitButton.textContent = "Save";
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((error) => {
+        return Promise.reject(`Error: ${error}`);
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.